File: conversation-websocket/node/outbound-call.ts
import twilio from "twilio";
import VoiceResponse from "twilio/lib/twiml/VoiceResponse";
import {
  customerPhoneNumber,
  ngrokUrl,
  twilioAccountSid,
  twilioAuthToken,
  twilioPhoneNumber,
} from "./env-vars";

const client = twilio(twilioAccountSid, twilioAuthToken);

async function makeCall() {
  const response = new VoiceResponse();
  response.connect().stream({
    url: `wss://${ngrokUrl}/ws`,
  });

  try {
    const call = await client.calls.create({
      twiml: response.toString(),
      to: customerPhoneNumber,
      from: twilioPhoneNumber,
    });

    console.log(`Call initiated with SID: ${call.sid}`);
  } catch (error) {
    console.error("Error making call:", error);
  }
}

makeCall();

File: conversation-websocket/python/outbound_call.py
import os
from pathlib import Path

from dotenv import load_dotenv
from twilio.rest import Client
from twilio.twiml.voice_response import Connect, VoiceResponse

load_dotenv(".env.local")

client = Client(os.environ["TWILIO_ACCOUNT_SID"], os.environ["TWILIO_AUTH_TOKEN"])


def make_call():
    response = VoiceResponse()
    connect = Connect()
    connect.stream(url=f"wss://{os.environ['NGROK_URL']}/ws")
    response.append(connect)

    client.calls.create(
        twiml=str(response),
        to=os.environ["CUSTOMER_PHONE_NUMBER"],
        from_=os.environ["TWILIO_PHONE_NUMBER"],
    )


if __name__ == "__main__":
    make_call()
